packages/x-date-pickers/src/PickersLayout/PickersLayout.tsx

```diff
@@ -0,0 +1,61 @@
+import * as React from 'react';
+import clsx from 'clsx';
+import { styled, useThemeProps } from '@mui/material/styles';
+import { PickersLayoutProps } from './PickersLayout.types';
+import { pickersLayoutClasses } from './pickersLayoutClasses';
+import usePickerLayout from './usePickerLayout';
+
+export const PickersLayoutRoot = styled('div', {
+  name: 'MuiPickersLayout',
+  slot: 'Root',
+  overridesResolver: (props, styles) => styles.root,
+})<{ ownerState: { isLandscape: boolean } }>(({ theme, ownerState }) => ({
+  display: 'grid',
+  gridAutoColumns: 'max-content auto max-content',
+  gridAutoRows: 'max-content auto max-content',
+
+  [`& .${pickersLayoutClasses.toolbar}`]: ownerState.isLandscape
+    ? {
+        gridColumn: theme.direction === 'rtl' ? 3 : 1,
+        gridRow: '1 / 3',
+      }
+    : { gridColumn: '1 / 4', gridRow: 1 },
+  [`& .${pickersLayoutClasses.actionBar}`]: { gridColumn: '1 / 4', gridRow: 3 },
+}));
+
+export const PickersLayoutContentWrapper = styled('div', {
+  name: 'MuiPickersLayout',
+  slot: 'Content',
+  overridesResolver: (props, styles) => styles.contentWrapper,
+})({
+  gridColumn: 2,
+  gridRow: 2,
+  display: 'flex',
+  flexDirection: 'column',
+});
+
+export const PickersLayout = React.forwardRef(function PickersLayout(
+  inProps: PickersLayoutProps<any, any>,
+  ref: React.Ref<HTMLDivElement>,
+) {
+  const props = useThemeProps({ props: inProps, name: 'MuiPickersLayout' });
+
+  const { toolbar, content, tabs, actionBar } = usePickerLayout(props);
+  const { sx, className, isLandscape } = props;
+
+  return (
+    <PickersLayoutRoot
+      ref={ref}
+      sx={sx}
+      className={clsx(className, pickersLayoutClasses.root)}
```

Comment:
Have you considered the following change aligning the component implementation with others and also applying `MuiPickersLayout-landscape` class in landscape orientation:

```diff
diff --git a/packages/x-date-pickers/src/PickersLayout/PickersLayout.tsx b/packages/x-date-pickers/src/PickersLayout/PickersLayout.tsx
index 4d2a8ac11..c4ee96b3c 100644
--- a/packages/x-date-pickers/src/PickersLayout/PickersLayout.tsx
+++ b/packages/x-date-pickers/src/PickersLayout/PickersLayout.tsx
@@ -1,10 +1,20 @@
 import * as React from 'react';
 import clsx from 'clsx';
 import { styled, useThemeProps } from '@mui/material/styles';
+import { unstable_composeClasses as composeClasses } from '@mui/utils';
 import { PickersLayoutProps } from './PickersLayout.types';
-import { pickersLayoutClasses } from './pickersLayoutClasses';
+import { getPickersLayoutUtilityClass, pickersLayoutClasses } from './pickersLayoutClasses';
 import usePickerLayout from './usePickerLayout';
 
+const useUtilityClasses = (ownerState: { isLandscape: boolean }) => {
+  const slots = {
+    root: ['root', ownerState.isLandscape && 'landscape'],
+    contentWrapper: ['contentWrapper'],
+  };
+
+  return composeClasses(slots, getPickersLayoutUtilityClass, {});
+};
+
 export const PickersLayoutRoot = styled('div', {
   name: 'MuiPickersLayout',
   slot: 'Root',
@@ -25,7 +35,7 @@ export const PickersLayoutRoot = styled('div', {
 
 export const PickersLayoutContentWrapper = styled('div', {
   name: 'MuiPickersLayout',
-  slot: 'Content',
+  slot: 'ContentWrapper',
   overridesResolver: (props, styles) => styles.contentWrapper,
 })({
   gridColumn: 2,
@@ -43,15 +53,17 @@ export const PickersLayout = React.forwardRef(function PickersLayout(
   const { toolbar, content, tabs, actionBar } = usePickerLayout(props);
   const { sx, className, isLandscape } = props;
 
+  const classes = useUtilityClasses({ isLandscape });
+
   return (
     <PickersLayoutRoot
       ref={ref}
       sx={sx}
-      className={clsx(className, pickersLayoutClasses.root)}
+      className={clsx(className, classes.root)}
       ownerState={{ isLandscape }}
     >
       {toolbar}
-      <PickersLayoutContentWrapper className={pickersLayoutClasses.contentWrapper}>
+      <PickersLayoutContentWrapper className={classes.contentWrapper}>
         {tabs}
         {content}
       </PickersLayoutContentWrapper>

```